tracim/views/core_api/session_controller.py

```diff
@@ -1,72 +1,47 @@
 # coding=utf-8
-import os
-from http.client import HTTPException
-
-from pyramid.httpexceptions import HTTPNoContent
-from pyramid.response import Response
-from sqlalchemy.orm.exc import NoResultFound
+from pyramid.config import Configurator
+try:  # Python 3.5+
+    from http import HTTPStatus
+except ImportError:
+    from http import client as HTTPStatus
 
 from tracim import TracimRequest
 from tracim.extensions import hapic
 from tracim.lib.core.user import UserApi
-from tracim.models.context_models import UserInContext
 from tracim.views.controllers import Controller
-from pyramid.config import Configurator
-
-from tracim.views import BASE_API_V2
-from tracim.views.core_api.schemas import UserSchema, NoContentSchema
+from tracim.views.core_api.schemas import UserSchema
+from tracim.views.core_api.schemas import NoContentSchema
 from tracim.views.core_api.schemas import LoginOutputHeaders
 from tracim.views.core_api.schemas import BasicAuthSchema
-from tracim.exceptions import NotAuthentificated, LoginFailed
-
-try:  # Python 3.5+
-    from http import HTTPStatus
-except ImportError:
-    from http import client as HTTPStatus
+from tracim.exceptions import NotAuthentificated
+from tracim.exceptions import AuthenticationFailed
 
 
 class SessionController(Controller):
 
     @hapic.with_api_doc()
     @hapic.input_headers(LoginOutputHeaders())
-    @hapic.input_query(BasicAuthSchema())
-    @hapic.handle_exception(LoginFailed, http_code=HTTPStatus.BAD_REQUEST)
+    @hapic.input_body(BasicAuthSchema())
+    @hapic.handle_exception(AuthenticationFailed, HTTPStatus.BAD_REQUEST)
     # TODO - G.M - 17-04-2018 - fix output header ?
     # @hapic.output_headers()
-    @hapic.output_body(
-        NoContentSchema(),
-        default_http_code=HTTPStatus.NO_CONTENT
-    )
+    @hapic.output_body(NoContentSchema(), default_http_code=HTTPStatus.NO_CONTENT)  # nopep8
     def login(self, context, request: TracimRequest, hapic_data=None):
         """
         Logs user into the system
         """
-        email = request.params['email']
-        password = request.params['password']
-        if not (email and password):
-            raise Exception
+        email = request.json_body['email']
```

Comment:
ask @buxx how to automatically fill an object from Schema in order not to have to fill yourself stuff from request.json_body.